crates/optimism/txpool/src/supervisor/errors.rs

```diff
@@ -116,7 +118,14 @@ pub enum InteropTxValidatorError {
 
 impl InteropTxValidatorError {
     /// Returns a new instance of [`RpcClientError`](Self::RpcClientError) variant.
-    pub fn client(err: impl error::Error + Send + Sync + 'static) -> Self {
+    pub fn client<T>(err: alloy_rpc_client::RpcError<T>) -> Self {
+        let err_msg = err.to_string();
+        
+        // Trying to parse the InvalidInboxEntry
+        if let Some(invalid_entry) = InvalidInboxEntry::parse_err_msg(&err_msg) {
+            return Self::InvalidInboxEntry(invalid_entry);
+        }
```

Comment:
this should work, I also do expect that this error msg parsing will be replaced by error codes instead